src/drawer.canvasentry.js

```diff
@@ -159,26 +163,41 @@ export default class CanvasEntry {
     }
 
     /**
-     * Utility function to handle wave color arguments
+     * Utility function to determine the type and handle the argument passed to
+     * a color setting method
      *
-     * When the color argument type is a string, it will be returned as is.
-     * Otherwise, it will be treated as an array, and a canvas gradient will
-     * be returned
+     * Return type is dependent on the color argument's type
      *
      * @since 5.3.0
      * @param {CanvasRenderingContext2D} ctx Rendering context of target canvas
-     * @param {string|string[]} color Fill color for the wave canvas, or an array of colors to apply as a gradient
-     * @returns {string|CanvasGradient} Returns a string fillstyle value, or a canvas gradient
+     * @param {string|string[]|CanvasImageSource} color A CSS color value as a
+     * string, an array of CSS color value strings, or a Canvas Image Source
+     * (HTML Image, SVG Image, HTML Video, HTML Canvas, Image Bitmap or
+     * Offscreen Canvas element)
+     * @returns {string|CanvasGradient|CanvasPattern} Returns a CSS color
+     * value string, a canvas gradient or a canvas pattern
      */
     getFillStyle(ctx, color) {
+        // if the color argument is a string, handle it as a CSS color value
         if (typeof color == 'string') {
             return color;
         }
 
-        const waveGradient = ctx.createLinearGradient(0, 0, 0, ctx.canvas.height);
-        color.forEach((value, index) => waveGradient.addColorStop((index / color.length), value));
+        // if it is an array, handle it as a gradient
+        if (Array.isArray(color)) {
+            const waveGradient = ctx.createLinearGradient(0, 0, 0, ctx.canvas.height);
+            color.forEach((value, index) => waveGradient.addColorStop((index / color.length), value));
+            return waveGradient;
+        }
+
+        // if it is an object, handle it as a CanvasImageSource
+        if (typeof color == "object") {
+            let pattern = ctx.createPattern(color, "repeat");
```

Comment:
is it necessary to create a variable here?
Could you please simply return the result of `createPattern`?
```
if (typeof color == "object") {
  return ctx.createPattern(color, "repeat");
}
```

Reply:
Also, I think, we need second argument of `createPattern`, the `repetition`, to be configurable
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern#parameters

Reply:
Updated to return the pattern instead of store it in a variable.

Regarding the configurable repetition parameter, please see my comment https://github.com/katspaugh/wavesurfer.js/pull/2348#discussion_r699497649